clang/lib/StaticAnalyzer/Checkers/NullabilityChecker.cpp

```diff
@@ -692,6 +692,14 @@ void NullabilityChecker::checkPreStmt(const ReturnStmt *S,
   NullConstraint Nullness = getNullConstraint(*RetSVal, State);
 
   Nullability RequiredNullability = getNullabilityAnnotation(RequiredRetType);
+  if (const auto *FunDecl = C.getLocationContext()->getDecl();
```

Comment:
Why isn't this logic added as a part of `getNullabilityAnnotation`? (I'm just curious, I can easily imagine that there is a good reason for that.)

Reply:
`getNullabilityAnnotation` is concerned only with the type. It is used both for return types and function parameters by multiple checkers.
Handling the function annotation there would mean I have to pass the function declaration (which might not be available in the client of the API), which is only relevant in one case (and maybe another couple in TrustNonnullChecker). That means I would need to complicate the function definition and potentially increase the PR footprint by touching all call sites.

Reply:
I see, thanks for the clarification.